google/ggrc-core
# BC: GGRC-7924 Complete selected Assessments

# Comment

**Should be merged to "asmt-bulk".**

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] #10063 

# Issue description

Add 'Complete' and 'Cancel' buttons on Bulk Complete modal.

# Steps to test the changes

Open Bulk Complete modal. Select some assessments. Click 'Select' button. LCA should be generated.
1) While LCA are not generated or if not all mandatory fields are filled, 'Complete' button should be disabled.
2) If all mandatory fields are filled, 'Complete' button should be enabled.
3) On 'Cancel' button, Bulk Complete modal should close.
4) On 'Complete' button:
If some attributes values was changed, confirm popup with next message `You've added new answers for custom attributes that will replace existing answers.` should be shown.
If click 'Proceed' on confirm popup, next  notifier message `Your bulk update is submitted. Once it is done you will get a notification. You can continue working with the app.` should be shown.
If bulk update is successfully completed next notifier message `Bulk update is finished successfully. Show results` should be shown. 'Show results' is link to the current page.
If bulk update is successfully completed with some errors next notifier message `There were some errors in bulk update. More details will be sent by email.` should be shown. (Needs approve from BA)
If bulk update is failed next notifier message `Bulk update is failed.` should be shown.
If withAuth() is failed next notifier message `Unable to Authorize` should be shown.
If some errors appears during the bulk update process, next notifier message `Bulk update failed. Please refresh the page and start bulk update again.` should be shown. (Needs approve from BA)

# Solution description

1) Add 'Complete' and 'Cancel' button.
2) When 'Complete' button is clicked, make auth BE for GDriver using withAuth()
method and send POST request to BE with assessments which should be completed in bulk.
3) Track background job like it's done for assessments generation.
4) When 'Cancel' button is clicked, close Bulk Complete modal.
5) Show necessary notifier messages.

# Sanity checklist


# PR Review checklist



# TODO

- [ ] Two notifier messages should be approved by BA.
